### PR TITLE
deps: update to tsc 3.1.1

### PR DESCRIPTION
**Pull request by brendankenny**
Typescript 3.1 is out, time to move off a nightly version and to the official release :)

Also removed a few outdated `@ts-ignore` comments.

### CODE REVIEW

File: typings/externs.d.ts [hunk 1]
```diff
@@ -69,7 +69,7 @@ declare global {
       cpuSlowdownMultiplier?: number
     }
 
-    export type Locale = 'en-US'|'en'|'en-AU'|'en-GB'|'en-IE'|'en-SG'|'en-ZA'|'en-IN'|'ar-XB'|'ar'|'bg'|'bs'|'ca'|'cs'|'da'|'de'|'el'|'en-XA'|'es'|'fi'|'fil'|'fr'|'he'|'hi'|'hr'|'hu'|'gsw'|'id'|'in'|'it'|'iw'|'ja'|'ko'|'lt'|'lv'|'mo'|'nl'|'nb'|'no'|'pl'|'pt'|'pt-PT'|'ro'|'ru'|'sk'|'sl'|'sr'|'sr-Latn'|'sv'|'ta'|'te'|'th'|'tl'|'tr'|'uk'|'vi'|'zh'|'zh-HK'|'zh-TW';
+    export type Locale = 'en-US'|'en'|'en-AU'|'en-GB'|'en-IE'|'en-SG'|'en-ZA'|'en-IN'|'ar-XB'|'ar'|'bg'|'bs'|'ca'|'cs'|'da'|'de'|'el'|'en-XA'|'es'|'fi'|'fil'|'fr'|'he'|'hi'|'hr'|'hu'|'gsw'|'id'|'in'|'it'|'iw'|'ja'|'ko'|'ln'|'lt'|'lv'|'mo'|'nl'|'nb'|'no'|'pl'|'pt'|'pt-PT'|'ro'|'ru'|'sk'|'sl'|'sr'|'sr-Latn'|'sv'|'ta'|'te'|'th'|'tl'|'tr'|'uk'|'vi'|'zh'|'zh-HK'|'zh-TW';
```
Comment (patrickhulce) [inline typings/externs.d.ts]: was it just missing before? seems unrelated :)

Comment (brendankenny) [reply typings/externs.d.ts]: > was it just missing before?

it was. Having `ln` as a key in `locales.js` but not here was the only new error when upgrading.

Not sure how it wasn't a problem before (but it might be related to the weird `en`/`en-US` bug that was `//@ts-ignore`d until this change)
